join.spec.md

```diff
@@ -100,204 +101,349 @@ flowchart TB
 
-Multiple {@join}s can be specified for different subgraphs. It is an error to {@join} an object against the same subgraph multiple times.
+There are several ways that the router can calculate a representation to pass to a subgraph. In this specification, all non-value types have a specific subgraph referred to as its "owner", specified via a `@join__owner` directive on the type. (Object types that are not value types are referred to as "entities"; the type `_Entity` referenced above is a union defined in each subgraph's schema consisting of the entity types defined by that subgraph. Only subgraphs which define entities need to define the `Query._entities` field.) The type must also have at least one `@join__type` directive specifying the owning subgraph along with a {key}. For each additional subgraph which can resolve fields returning that type, there should be exactly one `@join__type` directive specifying that subgraph along with a {key}, which should be identical to one of the keys specified with the owning subgraph.
 
-##! @join__owner
+A key is a set of fields on the type (potentially including sub-selections and inline fragments), specified as a string. If a type `T` is annotated with `@join__type(subgraph: G, key: "a b { c }")`, then it must be possible to resolve the full field set provided as a key on subgraph G. Additionally, if you take an object with the structure returned by resolving that field set and add a field `__typename: "T"`, then you should be able to pass the resulting value as a representation to the `Query._entities` field on subgraph G.
 
-Specify the graph which owns the object type.
+In order to resolve a field on an entity on the subgraph that owns its parent type, where that subgraph is different from the subgraph that resolved its parent object, the router first resolves a key for that object on the previous subgraph, and then uses that representation on the owning subgraph.
 
-```graphql definition
-directive @join__owner(graph: join__Graph!) on OBJECT
+For convenience, you may omit `@join__field(graph: A)` directives on fields whose parent type is owned by `A`.
+
+```graphql example -- Owned fields on owned types
+# Supergraph schema
+type Query {
+  fieldB: X @join__field(graph: B)
+}
+
+type X
+    @join__owner(graph: A)
+    # As the owner, A is allowed to have more than one key.
+    @join__type(graph: A, key: "x")
+    @join__type(graph: A, key: "y z")
+    # As non-owners, B and C can only have one key each and
+    # they must match a key from A.
+    @join__type(graph: B, key: "x")
+    @join__type(graph: C, key: "y z")
+  {
+  # Because B owns X, we can omit @join__field(graph: B)
+  # from these two fields.
+  x: String
+  y: String
+  z: String
+}
```

Comment:
Out of fear of the example growing too large, it would be good to see fields owned by B and/or C as well here.

Reply:
This is resolved in a more complex example below.

Reply:
Yep, this whole section is the "owned fields on owned types" section